examples/perplexity/perplexity.cpp

```diff
@@ -876,10 +876,12 @@ static void hellaswag_score(llama_context * ctx, const gpt_params & params) {
         // Compute log-probs in parallel
         // First we collect all tasks
         eval_pairs.clear();
+        eval_pairs.reserve((i1 - i0) * 4);
         for (size_t i = i0; i < i1; ++i) {
             auto & hs_cur = hs_data[i];
             size_t li = hs_cur.common_prefix;
             for (int s = 0; s < 4; ++s) {
+                eval_pairs.reserve((hs_cur.seq_tokens[s].size() - 1) - hs_cur.common_prefix);
                 for (size_t j = hs_cur.common_prefix; j < hs_cur.seq_tokens[s].size() - 1; j++) {
                     eval_pairs.emplace_back(hs_cur.i_batch + li++, hs_cur.seq_tokens[s][j + 1]);
                 }
```

Comment:
These reserves of eval_pairs aren't right - reserving in a loop without accounting for the existing capacity is wrong, and the initial reserve is trying to be precise but misses the innermost loop.